Slingshot.Api/Abstract/Repository.cs

```diff
@@ -130,6 +131,15 @@ public virtual string UserName
             return null;
         }
 
+#pragma warning disable 1998
+        public async virtual Task<bool> IsPrivate()
+        {
+            // Default we assume repo is public repo
+            _isPrivate = false;
```

Comment:
this is just an default value when children class do not implement it. 
if you look at the implementation in Bitbucket, _isPrivate is used as cache, which is what all the children class will do. later when we try to enable private repo for github, I will make this method to be "abstract" instead of "virtual", and will just throw a not implemented exception from here.